README.md

```diff
@@ -9,49 +8,163 @@ OMA Lightweight M2M IoT Agent: Overview
+The IoT Agent provides a mean to map Lightweight M2M objects supported by the
+client without the need to map them in the type or prevoprovision information.
+The mapping works as follows: whenever a device registration arrives to the IoT
+Agent **if there is no configured mapping for any of the objects supported by
+the decive** (that should appear in the registration payload), then **all the
+resources supported by the object in the OMA Registry** are configured **as
+passive resources** offered by the object.
+
+The OMA Registry information is read from two files: `omaRegistry.json` and
+`omaInverseRegistry.json`. This two files can be generated with the last
+information in the OMA Registry with the command `bin/downloadOmaRegistry.js`.
+Notice that the generated files **do not** have the same name than the original
+ones (so the result can be double-checked before changing them). In order to use
+the freshly downloaded ones, just remove the former and rename the latter.
+
+### Preprovisioning
+
+For individual provisioning of devices, LWM2M devices can be preprovisioned to
+the server, sending all the required information to the IoT Agent Provisioning
+API. Preprovisioned devices should target the standard '/rd' url instead of a
+type url. The preprovision configuration will be identified by the Endpoint name
+sent by the device.
+
+### A note about security
+
+IoT Agent security is still in development, so no Southbound or Northbound
+security mechanisms are provided. The NGSI Context Broker can be secured with a
+[PEP Proxy]() anyway, so the IoT Agent should be able to deal with token based
+security. This mechanism is achieved with the use of Keystone Trust Tokens. For
+more information on how to use them, please read the Security Section in
+[Node IoT Agent Library](https://github.com/telefonicaid/iotagent-node-lib).
+
+## Install
+
+Information about how to install the OMA Lightweight M2M IoT Agent can be found
+at the corresponding section of the
+[Installation & Administration Guide](https://fiware-iotagent-lwm2m.readthedocs.io/en/latest/administrationGuide).
+
+## Usage
+
+Information about how to use the IoT Agent can be found in the
+[User & Programmers Manual](https://fiware-iotagent-lwm2m.readthedocs.io/en/latest/userGuide).
+
+## API
+
+Apiary reference for the Configuration API can be found
+[here](http://docs.telefonicaiotiotagents.apiary.io/#reference/configuration-api).
+More information about IoT Agents and their APIs can be found in the IoT Agent
+Library [documentation](https://iotagent-node-lib.rtfd.io/).
+
+## Quality Assurance
```

Comment:
Same comment to the one published at https://github.com/telefonicaid/iotagent-ul/pull/301#discussion_r236292514

Reply:
> This section should include:
> 
> Information about the last published version (so the user can know the gap between "version tested" and the newest one, evaluating if the information is obsolete enough for him or not)
>
> A link to the labeling calculation procedure, i.e. which metrics and in which form decide which label to assign. Ideally, this should be a mathematical formula/table leaving no room for subjective interpretation.
>
> A link (or several) to the report(s) in which the labels are based, explaining which particular values of the metrics have been used (and how they has been got) in the above formula.
>
> In addition, why two of them are in "no results" status?

All these are good suggestions, but I am unable to implement the requested changes at the moment.

The badges are dynamically driven from the data held in a series of JSON files. The data you are requesting is currently not present within FIWARE. The existing files only hold the **version tested** and not the **current version.** , FIWARE QA don't hold the current version for this component -that is under Telefonica's control

Regarding the testing procedure, the following links within the FIWARE catalogue describe the process:

* https://github.com/Fiware/catalogue#testing
* https://www.fiware.org/developers/qa-labeling-of-fiware-components/

It is not part of the current [contribution requirements](https://github.com/Fiware/contribution-requirements) to display additional links and information - I agree that the QA section could be expanded sometime, but the limited information within this block is currently just a summary of the parts which are available. If you want to display more that is fine, but it would make sense to add a PR to the requirements so that all enablers display the same information consistently.

The two missing ("no results" status) badges are because the QA data has not been updated yet.

Reply:
To avoid duplication :) please see https://github.com/telefonicaid/iotagent-json/pull/341/files#r240975502